frontend/src/metabase/setup/components/UserStep/UserStep.tsx

```diff
@@ -44,7 +58,8 @@ export const UserStep = ({ stepLabel }: NumberedStepProps): JSX.Element => {
         </StepDescription>
       )}
       <UserForm
-        user={user}
+        user={isHosted ? cloudUser : user}
```

Comment:
can we set the user from the URL as initial state on redux instead?
That way we're sure we only have one user object which is always in sync.
For example reading this PR, i'm not sure if we store the correct email on redux, which will be used in the newsletter step later on

Reply:
Makes sense.
Thank you!